rskj-core/src/main/java/co/rsk/rpc/Web3InformationRetriever.java

```diff
@@ -0,0 +1,134 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.rpc;
+
+import co.rsk.core.bc.AccountInformationProvider;
+import co.rsk.db.RepositoryLocator;
+import org.ethereum.core.Block;
+import org.ethereum.core.Blockchain;
+import org.ethereum.core.Transaction;
+import org.ethereum.core.TransactionPool;
+import org.ethereum.rpc.exception.RskJsonRpcRequestException;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.ethereum.rpc.TypeConverter.stringHexToBigInteger;
+import static org.ethereum.rpc.exception.RskJsonRpcRequestException.invalidParamError;
+
+/**
+ * Retrieves information requested by web3 based on the block identifier:
+ * <p>
+ * HEX String  - an integer block number
+ * String "earliest"  for the earliest/genesis block
+ * String "latest"  - for the latest mined block
+ * String "pending"  - for the pending state/transactions
+ */
+public class Web3InformationRetriever {
+    private final TransactionPool transactionPool;
+    private final Blockchain blockchain;
+    private final RepositoryLocator locator;
+
+    private static final String EARLIEST = "earliest";
+    private static final String LATEST = "latest";
+    private static final String PENDING = "pending";
+
+
+    public Web3InformationRetriever(TransactionPool transactionPool, Blockchain blockchain, RepositoryLocator locator) {
+
+        this.transactionPool = transactionPool;
+        this.blockchain = blockchain;
+        this.locator = locator;
+    }
+
+    /**
+     * Retrieves the block based on the identifier.
+     * @param identifier {@link Web3InformationRetriever}
+     * @return An optional containing the block if found.
+     * @throws RskJsonRpcRequestException if the identifier is {@link #PENDING}, as it is not implemented yet.
+     */
+    public Optional<Block> getBlock(String identifier) {
+        if (PENDING.equals(identifier)) {
+            throw RskJsonRpcRequestException.unimplemented("This method doesn't support 'pending' as a parameter");
+        }
+
+        Block block;
+        if (LATEST.equals(identifier)) {
+            block = blockchain.getBestBlock();
+        } else if (EARLIEST.equals(identifier)) {
+            block = blockchain.getBlockByNumber(0);
+        } else {
+            block = this.blockchain.getBlockByNumber(getBlockNumber(identifier));
+        }
+
+        return Optional.ofNullable(block);
+    }
+
+    /**
+     * Retrieves an {@link AccountInformationProvider} based on the identifier.
+     * @param identifier {@link Web3InformationRetriever}
+     * @return The {@link AccountInformationProvider}
+     * @throws RskJsonRpcRequestException if the block indicated by the identifier is not found or the state for the
+     * block was not found.
+     */
+    public AccountInformationProvider getState(String identifier) {
```

Comment:
What about renaming `getState` to `getInformationProvider`? I think **state** is a huge concept in our domain